internal/objectstore/fileobjectstore.go

```diff
@@ -387,63 +418,75 @@ func (t *fileObjectStore) loop() error {
 			timer.Reset(defaultPruneInterval)
 
 			if err := t.prune(ctx, t.list, t.deleteObject); err != nil {
-				t.logger.Errorf(context.TODO(), "prune: %v", err)
+				t.logger.Errorf(ctx, "prune: %v", err)
 				continue
 			}
 		}
 	}
 }
 
-func (t *fileObjectStore) get(ctx context.Context, path string) (io.ReadCloser, int64, error) {
-	t.logger.Debugf(context.TODO(), "getting object %q from file storage", path)
+func (t *fileObjectStore) get(ctx context.Context, path string, fallbackStrategy FallbackStrategy) (io.ReadCloser, int64, error) {
```

Comment:
Do we foresee more than the one `RemoteFallback`  (& `NoFallback` of course) being added in the future?

If not, these methods are reduced to methods with a bool param, which can be considered an anti-pattern.

This is a nitpick, but an alternative would be to write `get` and `getRemote` where `getRemote` first calls `get` (or something similar). What do you think?

Reply:
We don't always want to fallback to a remote, only for the currently implemented requests (get and getSHA256Prefix).

> If not, these methods are reduced to methods with a bool param, which can be considered an anti-pattern.

Except they're more descriptive and the default value isn't a footgun :wink: 

> This is a nitpick, but an alternative would be to write get and getRemote where getRemote first calls get (or something similar). What do you think?

I'll be happy to look, but it might increase complexity because we don't require it for all get requests.

Reply:
This isn't an important concern really, happy to keep as is